File: product/src/main/kotlin/org/platypus/impl/module/product/models/ProductPriceHistoryModel.kt
package org.platypus.impl.module.product.models

import org.platypus.impl.korm.model.ModelNamed
import org.platypus.impl.module.product.entities.ProductPriceHistory

object ProductPriceHistoryModel : ModelNamed<ProductPriceHistoryModel, ProductPriceHistory>("product.price.history")

File: product/src/main/kotlin/org/platypus/impl/module/product/models/ProductPriceHistoryModel.py
class ProductPriceHistory(models.Model):
    """ Keep track of the ``product.domModule`` standard prices as they are changed. """
    _name = 'product.price.history'
    _rec_name = 'datetime'
    _order = 'datetime desc'

    def _get_default_company_id(self):
        return self._context.get('force_company', self.env.user.company_id.id)

    company_id = fields.Many2one('res.company', string='Company',
                                 default=_get_default_company_id, required=True)
    product_id = fields.Many2one('product.product', 'Product', ondelete='cascade', required=True)
    datetime = fields.Datetime('Date', default=fields.Datetime.now)
    cost = fields.Float('Cost', digits=dp.get_precision('Product Price'))